src/components/Dropdown/component.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { useState, useRef, useEffect } from 'react'
+import { ChevronDown, X } from 'lucide-react';
+
+interface Option {
+    value: string;
+    label: string;
+    disabled?: boolean;
+}
+
+interface DropdownProps {
+    options: Option[];
+    multiSelect?: boolean;
+    placeholder?: string;
+    onChange: (selected: string | string[]) => void; //onchange 
+    customOptionRenderer?: (option: Option) => React.ReactNode;
+    disabled?: boolean
+}
+const DropDown: React.FC<DropdownProps> = ({
+    options,
+    multiSelect = false,
+    placeholder = 'Select...',
+    onChange,
+    customOptionRenderer,
+    disabled = false,
+}) => {
+    const [isOpen, setIsOpen] = useState(false);
+    const [selected, setSelected] = useState<string[]>([]);
+    const [searchTerm, setSearchterm] = useState('');
+    const dropdownRef = useRef<HTMLDivElement>(null);
+    const [dropdownHeight, setDropdownHeight] = useState(0);
+
+    useEffect(() => { //
+        if (isOpen && dropdownRef.current) {
+            setDropdownHeight(dropdownRef.current.scrollHeight);
+        } else {
+            setDropdownHeight(0);
+        }
+    }, [isOpen])
+
+    const filteredOptions = options.filter(option =>
+        option.label.toLowerCase().includes(searchTerm.toLowerCase())
+    );
+
+    const handleSelect = (value: string) => {
+        if (multiSelect) { //toggle the selected state of the option 
+            const updatedSelected = selected.includes(value)
+                ? selected.filter(item => item !== value)//  assign everything but that equal to the option value
+                : [...selected, value]; //take the pervious values and add option value to the end
+            setSelected(updatedSelected)
+            onChange(updatedSelected)
+        } else {
+            setSelected([value]);
+            onChange(value); // hoist value up to parent component
+            setIsOpen(false);
+        }
+    };
+
+    const handleClear = () => {
+        setSelected([]); // clear the state first 
+        onChange(multiSelect ? [] : ''); // pass an array if multi-select option is on but pass a string if not 
+    }
+
+    const handleKeyDown = (e: React.KeyboardEvent) => {
+        if (e.key === 'ArrowDown' || e.key === 'Arrow up') {
+            e.preventDefault(); //preveny default function which may scrolling the page 
+            const currentIndex = filteredOptions.findIndex(options => options.value === selected[0])
+            const nextIndex = e.key === 'ArrowDown'
```

Comment:
I see that this is getting a nextIndex, but I don't see where it's doing anything? Does this work?